### PR TITLE
Make it possible to use sparse_categorical_crossentropy with variable number of classes in each batch (TensorFlow backend)

### PR DESCRIPTION
**Pull request by rkadlec**
Reshape now uses dynamic shape information from tf.shape(output) instead of static output.get_shape().
This makes it possible to have a different number of target classes in each batch.

### CODE REVIEW

**Comment by fchollet** [comment]
This needs tests. Also need to check behavior consistency with Theano. Add to `backend_test.py`.

**Comment by rkadlec** [comment]
What kind of a test would be appropriate? Just compute softmax over two different batches? Is there some test that I can use as inspiration?

**Comment by fchollet** [comment]
Sorry for the delay, I missed your answer.

Basically add a test in `backend_test.py` where you compute a softmax over different batches and check consistency with Theano (same as every other backend test).

**Comment by ShayanPersonal** [comment]
This fixed an issue I've been having with sparse_categorical_crossentropy. Thanks.
